src/Common/LRUCache.h

```diff
@@ -326,13 +377,33 @@ class LRUCache
             }
 
             const auto & cell = it->second;
+            auto evict_status = evict_policy.canRelease(*cell.value);// in default, it is CAN_EVITCT
+            if (evict_status == LRUCacheEvictStatus::CAN_EVITCT)
+            {
+                // always call release() before erasing an element
+                // in default, it's an empty action
+                evict_policy.release(*cell.value);
 
-            current_size -= cell.size;
-            current_weight_lost += cell.size;
+                current_size -= cell.size;
+                current_weight_lost += cell.size;
 
-            cells.erase(it);
-            queue.pop_front();
-            --queue_size;
+                cells.erase(it);
+                queue.pop_front();
+                --queue_size;
+            }
+            else if (evict_status == LRUCacheEvictStatus::SKIP_EVICT)
+            {
+                // skip this element and try to evict the remaining ones.
+                key_it++;
+                continue;
+            }
+            else if (evict_status == LRUCacheEvictStatus::TERMINATE_EVICT)
+            {
+                // maybe we want to stop this iteration once we meet the first unreleasable element
+                break;
+            }
+            LOG_ERROR(&Poco::Logger::get("LRUCache"), "This condition branch should not be reached.");
```

Comment:
Better to use `switch` without default branch, so there will be no need in this LOG_ERROR and abort()